code/game/objects/items/storage/backpack.dm

```diff
@@ -365,6 +365,64 @@
 	icon_state = "smock"
 	storage_type = /datum/storage/backpack/no_delay
 
+/obj/item/storage/backpack/marine/duffelbag
+	name = "\improper TGMC Duffelbag"
+	desc = "A massive bag designed for those interested in being the squad Mule. \
+	Features extra storage space at the cost of speed. \
+	Any squadmates can easily access the storage with right-click"
+	icon_state = "duffel"
+	worn_icon_state = "duffel"
+	storage_type = /datum/storage/backpack/duffelbag
+	slowdown = 0.2
+
+/obj/item/storage/backpack/marine/duffelbag/equipped(mob/equipper, slot)
+	. = ..()
+	if(slot == SLOT_BACK)
+		RegisterSignal(equipper, COMSIG_CLICK_RIGHT, PROC_REF(on_rclick_duffel_wearer))
+		for(var/mob/M AS in storage_datum.content_watchers)
+			storage_datum.close(M)
+
+/obj/item/storage/backpack/marine/duffelbag/unequipped(mob/unequipper, slot)
+	. = ..()
+	UnregisterSignal(unequipper, COMSIG_CLICK_RIGHT)
+
+///Allows non-wearers to access this inventory
+/obj/item/storage/backpack/marine/duffelbag/proc/on_rclick_duffel_wearer(datum/source, mob/clicker)
+	SIGNAL_HANDLER
+	if(clicker == loc) //Wearer can't open this inventory
+		return
+	storage_datum.open(clicker)
```

Comment:
Does this uh, have a range check or anything else?
Does this close when the wearer moves?

Reply:
Storages close when you move, that behaviour doesn't change when you strap the storage to a mob.
I uh, forgot to check adjacent so I'll throw that in